### PR TITLE
[feature](multi-catalog) Support es datasource

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Support es datasource, This PR supports query index related metadata.

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalDataSource.java [hunk 1]
```diff
@@ -18,42 +18,131 @@
...
      * Default constructor for EsExternalDataSource.
      */
-    public EsExternalDataSource(String name, Map<String, String> props) {
+    public EsExternalDataSource(String name, Map<String, String> props) throws DdlException {
         setName(name);
         getDsProperty().setProperties(props);
         setType("es");
+        validate(props);
+        dbNameToId.put("user", nextId.incrementAndGet());
+        dbNameToId.put("system", nextId.incrementAndGet());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalDataSource.java:81]: the dbId should be global unique id. So we need to use Catalog.getNextId();
And add comment to explain this 2 databases.

**Comment by stalary** [comment]
Currently, it only runs es7/8 simple types, es6 and array types need to be developed later.

File: fe/fe-core/pom.xml [hunk 2]
```diff
@@ -629,6 +629,7 @@ under the License.
         <dependency>
             <groupId>org.apache.iceberg</groupId>
             <artifactId>iceberg-core</artifactId>
+            <scope>provided</scope>
```
Comment (morningman) [inline fe/fe-core/pom.xml:632]: Are you sure that this dependency is provided?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalDataSource.java [hunk 3]
```diff
@@ -18,43 +18,175 @@
...
+        try {
+            validate(this.dsProperty.getProperties());
+        } catch (DdlException e) {
+            LOG.warn("validate error", e);
+        }
+        dbNameToId = Maps.newConcurrentMap();
+        idToDb = Maps.newConcurrentMap();
+        this.esRestClient = new EsRestClient(this.nodes, this.username, this.password, this.enableSsl);
+        long defaultDbId = Catalog.getCurrentCatalog().getNextId();
+        dbNameToId.put("default", defaultDbId);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalDataSource.java:157]: define a `static final` field for this "default" string.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/external/EsExternalTable.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        return dbName;
+    }
+
+    @Override
+    public TTableDescriptor toThrift() {
+        return super.toThrift();
+    }
+
+    public EsTable toEsTable() {
+        EsTable esTable = new EsTable(this.id, this.name, this.getFullSchema(), TableType.ES_EXTERNAL_TABLE, name, null,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/external/EsExternalTable.java:114]: I think we don't need to new a `EsTable` everytime we call `toEsTable()`.
You can create it in `init`.

File: fe/fe-core/pom.xml [hunk 5]
```diff
@@ -624,6 +624,7 @@ under the License.
...
```
Comment (morningman) [inline fe/fe-core/pom.xml:627]: remove this

**Review by morningman — DISMISSED** [review]
LGTM
please rebase to resolve the conflict

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**